Update Backend docs (#14840)
PowerDNS/pdns: docs/backends/random.rst

==============

.. deprecated:: 4.6.0
  This backend was removed in 4.6.0

* Native: Yes
* Primary: No
* Secondary: No
* Producer: No
* Consumer: No
* Autosecondary: No
* DNS Update: No
* DNSSEC: Yes, no key storage
* Disabled data: No
* Comments: No

> * API: ???

Reviewer: ```suggestion
* API: Read-only
```

> * Multiple instances: ???

Reviewer: ```suggestion
* Multiple instances: No
```

* Zone caching: No
* Module name: random
* Launch: ``random``

> This is a very silly backend which is discussed in the :doc:`Backends

Reviewer: Nitpick: if you have changed the tense of the introduction sentence to past tense, you need to change this sentence, too.
```suggestion
This used to be a very silly backend which was discussed in the :doc:`Backends
```

.. deprecated:: 4.6.0

>   This backend has been removed in 4.6.0

Reviewer 2: Not technically in scope, but if you don't mind...
```suggestion
  This backend was been removed in 4.6.0
```

Reviewer 2: err, yeah, that.
